lib/streamlit/state/session_state.py

```diff
@@ -376,18 +380,31 @@ def as_widget_states(self) -> List[WidgetStateProto]:
         return self._new_widget_state.as_widget_states()
 
 
+_state_use_warning_already_displayed = False
+
+
 def get_session_state() -> SessionState:
     """Get the SessionState object for the current session.
 
     Note that in streamlit scripts, this function should not be called
     directly. Instead, SessionState objects should be accessed via
     st.session_state.
     """
+    global _state_use_warning_already_displayed
     from streamlit.report_thread import get_report_ctx
 
     ctx = get_report_ctx()
 
-    assert ctx is not None
+    # If there is no report context because the script is run bare, have
+    # session state act as an always empty dictionary, and print a warning.
+    if ctx is None:
+        if not _state_use_warning_already_displayed:
+            _state_use_warning_already_displayed = True
+            if not st._is_running_with_streamlit:
+                logger.warning(
+                    "Session state does not function when running a script without `streamlit run`"
+                )
+        return {}  # type: ignore
```

Comment:
Actually, hm, I just realized that I think we may have to return a new `SessionState` object here instead of an empty dict since `LazySessionState` could also access properties like `filtered_state` as well as things in attribute style that won't exist on a normal dictionary.